modules/apps/headless/headless-site/headless-site-test/src/testIntegration/java/com/liferay/headless/site/resource/v1_0/test/SiteResourceTest.java

```diff
@@ -987,6 +1034,81 @@ private void _testPutSiteBatch() throws Exception {
 				updatedSite.getExternalReferenceCode()));
 	}
 
+	private void _testPutSiteBatchSiteParentSiteERC() throws Exception {
+		try (SafeCloseable safeCloseable =
+				LazyReferencingThreadLocal.setEnabledWithSafeCloseable(true)) {
+
+			Site postParentSite = testPutSite_addSite();
+
+			Site postSite = testPutSite_addSite();
+
+			Site randomSite = randomSite();
+
+			randomSite.setExternalReferenceCode(
+				postSite.getExternalReferenceCode());
+
+			randomSite.setParentSiteERC(
+				postParentSite.getExternalReferenceCode());
+
+			Site putSite = siteResource.putSite(randomSite);
+
+			Assert.assertEquals(StringPool.BLANK, putSite.getParentSiteERC());
```

Comment:
Shouldn't this result in a matching ERC? From my reading the parent site exists.

In any case, we should have a test that confirms the ERC was set while lazy reference is enabled.